compile/compile.go

```diff
@@ -234,7 +234,7 @@ func addEntrypointsFromAnnotations(c *Compiler, ar []*ast.AnnotationsRef) error
 				}
 			case "rule":
 				if r := ref.GetRule(); r != nil {
-					entrypoint = r.Ref()
+					entrypoint = r.Path()
```

Comment:
`Rule.Path()` is deprecated; but here we have a clear use for a rule path that only includes ground.

So the question is, should `Path()` be un-deprecated with an updated godoc; or should we perhaps duplicate the behavior under a different name to make mistakes less likely?

Reply:
Can't we use `r.Ref().GroundPrefix()` ?

Reply:
Or you can directly use the `Extend` call?

Reply:
@ashutosh-narkar That's actually what `Rule.Path` does, I think ([ast/policy source](https://github.com/open-policy-agent/opa/blob/4bc74912372fd5479cb8700074582723316d3b06/ast/policy.go#L600-L608)): it wraps an `Extend` call around the ground prefix of the ref.

Reply:
Re: GroundPrefix() -- as long as we check edge cases like this
```rego
p.foo.bar if true
p.foo[x] if x := 12
```
we should be OK there 🤔